tools/sst_dump_tool.cc

```diff
@@ -219,63 +245,59 @@ int SstFileDumper::ShowAllCompressionSizes(
-      if ((compressed_blocks + not_compressed_blocks) > num_data_blocks) {
-        num_data_blocks = compressed_blocks + not_compressed_blocks;
-      }
-      const uint64_t ratio_not_compressed_blocks =
-          (num_data_blocks - compressed_blocks) - not_compressed_blocks;
-      const double compressed_pcnt =
-          (0 == num_data_blocks) ? 0.0
-                                 : ((static_cast<double>(compressed_blocks) /
-                                     static_cast<double>(num_data_blocks)) *
-                                    100.0);
-      const double ratio_not_compressed_pcnt =
-          (0 == num_data_blocks)
-              ? 0.0
-              : ((static_cast<double>(ratio_not_compressed_blocks) /
-                  static_cast<double>(num_data_blocks)) *
-                 100.0);
-      const double not_compressed_pcnt =
-          (0 == num_data_blocks)
-              ? 0.0
-              : ((static_cast<double>(not_compressed_blocks) /
-                  static_cast<double>(num_data_blocks)) *
-                 100.0);
-      fprintf(stdout, " Compressed: %6" PRIu64 " (%5.1f%%)", compressed_blocks,
-              compressed_pcnt);
-      fprintf(stdout, " Not compressed (ratio): %6" PRIu64 " (%5.1f%%)",
-              ratio_not_compressed_blocks, ratio_not_compressed_pcnt);
-      fprintf(stdout, " Not compressed (abort): %6" PRIu64 " (%5.1f%%)\n",
-              not_compressed_blocks, not_compressed_pcnt);
-    } else {
-      fprintf(stdout, "Unsupported compression type: %s.\n", i.second);
-    }
+  std::string column_family_name;
+  int unknown_level = -1;
+  TableBuilderOptions tb_opts(
+      imoptions, moptions, ikc, &block_based_table_factories, compress_type,
+      0 /* sample_for_compression */, compress_opt,
+      false /* skip_filters */, column_family_name, unknown_level);
+  uint64_t num_data_blocks = 0;
+  auto start = std::chrono::steady_clock::now();
+  uint64_t file_size =
+      CalculateCompressedTableSize(tb_opts, block_size, &num_data_blocks);
+  auto end = std::chrono::steady_clock::now();
+  auto duration = std::chrono::duration_cast<std::chrono::microseconds>(end - start);
+  fprintf(stdout, " Size: %10" PRIu64, file_size);
+  fprintf(stdout, " Blocks: %6" PRIu64, num_data_blocks);
+  fprintf(stdout, " Time Taken: %10s microsecs",
+          std::to_string(duration.count()).c_str());
+  const uint64_t compressed_blocks =
+       opts.statistics->getAndResetTickerCount(NUMBER_BLOCK_COMPRESSED);
+  const uint64_t not_compressed_blocks =
+       opts.statistics->getAndResetTickerCount(NUMBER_BLOCK_NOT_COMPRESSED);
+  // When the option enable_index_compression is true,
+  // NUMBER_BLOCK_COMPRESSED is incremented for index block(s).
+  if ((compressed_blocks + not_compressed_blocks) > num_data_blocks) {
+     num_data_blocks = compressed_blocks + not_compressed_blocks;
   }
+  const uint64_t ratio_not_compressed_blocks =
+       (num_data_blocks - compressed_blocks) - not_compressed_blocks;
```

Comment:
This formula looks wrong (it'll always evaluate to 0?). What exactly are we trying to calculate here?

Reply:
From the code it seems "num_data_blocks = compressed_blocks + not_compressed_blocks"  only when 
"compressed_blocks + not_compressed_blocks > num_data_blocks" so that
ratio_not_compressed_blocks will be 0 instead of -ve value.

Also, its mentioned that 
//When the option enable_index_compression is true,
// NUMBER_BLOCK_COMPRESSED is incremented for index block(s).
so I guess in that case, compressed_blocks increases which makes the total sum > num_data_blocks.

When I run the tool, for instance, num_data_blocks = 3320, compressed_blocks = 3321, non_compressed_blocks = 0 (before changing the num_data_blocks value)

Reply:
Interesting. Thanks for digging into this. Its still not clear when "compressed_blocks + not_compressed_blocks < num_data_blocks", which is when "ratio_not_compressed_blocks" will be > 0. Anyways, since it was there before, its a separate issue and we can probably follow it up outside this PR.